util/comparator.cc

```diff
@@ -216,6 +218,72 @@ class ReverseBytewiseComparatorImpl : public BytewiseComparatorImpl {
     return -a.compare(b);
   }
 };
+
+// EXPERIMENTAL
+// Comparator with 64-bit integer timestamp.
+// We did not performance test this yet.
+template <typename T>
+class ComparatorWithU64TsImpl : public Comparator {
+  static_assert(std::is_base_of<Comparator, T>::value,
+                "T must be a inherited type of comparator");
+
+ public:
+  ComparatorWithU64TsImpl()
+      : Comparator(/*ts_sz=*/sizeof(uint64_t)),
+        cmp_without_ts_(BytewiseComparator()) {
+    assert(cmp_without_ts_);
+    assert(cmp_without_ts_->timestamp_size() == 0);
+  }
+  static const char* kClassName() {
+    // Putting this up just for getting opinion,  need to make this better.
+    static char buf[128];
+    sprintf(buf, "%s.u64", T::kClassName());
```

Comment:
We discourage the usage of `sprint`. When you have to, use stringstream, especially when perf is not an issue here.

Reply:
thanks for the pointer modified it